src/main/java/frc/robot/commands/ArmAutoCommand.java

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.ArmSubsystem;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import java.lang.Math;
+
+public class ArmAutoCommand extends CommandBase {
+  private static final double firstArmLength = 110;
+  private static final double secondArmLength = 94;
+
+  private final ArmSubsystem armSubsystem;
+  private double firstArmAngle;
+  private double secondArmAngle;
+  private double currentX;
+  private double currentY;
+
+  public ArmAutoCommand(ArmSubsystem armSubsystem, double x, double y) {
+    this.armSubsystem = armSubsystem;
+    getAngles(x, y);
+    currentX = x;
+    currentY= y;
+    addRequirements(armSubsystem);
+  }
+  private void getAngles(double x, double y) {
+    //Needed to calculate the first and second arm angles
+    double prelimAngle = (
+      -1*Math.acos(
+        ( (x*x) + (y*y) - (firstArmLength*firstArmLength) - (secondArmLength*secondArmLength) )/(2*firstArmLength*secondArmLength)
+      )
+    );
+    //Get angle of the first arm (relative to a flat bottom line) (in radians)
+    firstArmAngle = (
+            Math.atan(y/x)-Math.atan((secondArmLength*Math.sin(prelimAngle) )/( (firstArmLength+(secondArmLength*Math.cos(prelimAngle))))
+    ));
+    //Get angle of the second arm relative to the first one (in radians)
+    secondArmAngle = prelimAngle+firstArmAngle;
+  }
+  public static double getPrelimAngle(double x, double y) {
+    return (
+      -1*Math.acos(
+        ( (x*x) + (y*y) - (firstArmLength*firstArmLength) - (secondArmLength*secondArmLength) )/(2*firstArmLength*secondArmLength)
+      )
+    );
+  }
+  public static double getBaseAngle(double x, double y) {
+    double pAngle = (
+      -1*Math.acos(
+        ( (x*x) + (y*y) - (firstArmLength*firstArmLength) - (secondArmLength*secondArmLength) )/(2*firstArmLength*secondArmLength)
+      )
+    );
+    double fArmAngle = (
+            Math.atan(y/x)-Math.atan((secondArmLength*Math.sin(pAngle) )/( (firstArmLength+(secondArmLength*Math.cos(pAngle))))
+    ));
+    return fArmAngle;
+  }
+  
+  public static double getElbowAngle(double x, double y) {
+    double pAngle = (
+      -1*Math.acos(
+        ( (x*x) + (y*y) - (firstArmLength*firstArmLength) - (secondArmLength*secondArmLength) )/(2*firstArmLength*secondArmLength)
+      )
+    );
+    double fArmAngle = (
+            Math.atan(y/x)-Math.atan((secondArmLength*Math.sin(pAngle) )/( (firstArmLength+(secondArmLength*Math.cos(pAngle))))
+    ));
+    return fArmAngle+pAngle;
```

Comment:
This returns the composite angle of the base arm + elbow arm, but the encoder will only track the elbow arm's angle without the base arm's component. Actual setpoint might just be pAngle, though I'm not certain of that.

Reply:
With the way the math works, pAngle returns a value relative to the first arm angle, so the first arm angle needs to be added to it to make it relative to the ground. If this is confusing, I can explain it a bit more in person, but the actual setpoint needs to be the two added together for it to work.